lib/dl_connector_greenplum/dl_connector_greenplum/core/adapters.py

```diff
@@ -72,3 +77,20 @@ class AsyncGreenplumAdapter(AsyncPostgresAdapter):
 
     async def get_table_info(self, table_def: TableDefinition, fetch_idx_info: bool) -> RawSchemaInfo:
         raise NotImplementedError()
+
+    @asynccontextmanager
+    async def _query_preparation_context(self, connection: asyncpg.Connection) -> AsyncIterator[None]:
+        async with super()._query_preparation_context(connection):
+            db_version = await connection.fetchval("SELECT version()")
+            if "greenplum" not in db_version.lower():
```

Comment:
do we really need it in GP Adapter?

Reply:
Unfortunately we do
We use postgres in greenplum tests at the moment and I believe that some users rely on this behaviour (not that it is okay, it's not, just saying)